app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java

```diff
@@ -0,0 +1,275 @@
+        view.findViewById(R.id.btnAdd).setOnClickListener(v -> setupAdd(view));
+    }
+
+    private void setupAdd(View view) {
+        String nickname=((EditText)view.findViewById(R.id.etNickname)).getText().toString();
+        String number=((EditText) view.findViewById(R.id.etPhoneNumber)).getText().toString();
+        EmergencyContacts contact = new EmergencyContacts();
+        if(nickname.isEmpty() && number.isEmpty()){
+            Toast.makeText(getContext(),"Please, fill all the fields",Toast.LENGTH_SHORT).show();
+        }
+        else{
+            contact.put(EmergencyContacts.USER_KEY,currentUser);
+            contact.put(EmergencyContacts.NICKNAME_KEY,nickname);
+            contact.put(EmergencyContacts.NUMBER_KEY,new Long(number));
+            contact.saveInBackground(e -> {
+                if(e!=null){
+                    Log.e(TAG, "setupAdd: Issue adding contact ",e );
+                }
+                Toast.makeText(getContext(),"Contact saved!!", Toast.LENGTH_SHORT).show();
+                CardView cvNewContact = view.findViewById(R.id.cvNewContact);
+                ImageButton btnAddContact = view.findViewById(R.id.btnAddContact);
+                cvNewContact.setVisibility(View.GONE);
+                btnAddContact.setRotation(0);
+                btnAddContact.setColorFilter(ContextCompat.getColor(getContext(), R.color.blue_cute), android.graphics.PorterDuff.Mode.MULTIPLY);
+                adapter.addFirst(contact);
+                ((RecyclerView)view.findViewById(R.id.rvEmergencyContacts)).smoothScrollToPosition(0);
+            });
+        }
+    }
+
+    private void setupAddContact(View view) {
+        CardView cvNewContact = view.findViewById(R.id.cvNewContact);
+        ImageButton btnAddContact = view.findViewById(R.id.btnAddContact);
+        if (cvNewContact.getVisibility() == View.VISIBLE) {
+            cvNewContact.setVisibility(View.GONE);
+            btnAddContact.setRotation(0);
+            btnAddContact.setColorFilter(ContextCompat.getColor(getContext(), R.color.blue_cute), android.graphics.PorterDuff.Mode.MULTIPLY);
+        } else {
+            cvNewContact.setVisibility(View.VISIBLE);
+            btnAddContact.setRotation(45);
+            btnAddContact.setColorFilter(ContextCompat.getColor(getContext(), R.color.red_ss), android.graphics.PorterDuff.Mode.MULTIPLY);
+        }
+    }
+
+    private void putProfileImage() {
+        int radius = 100000;
+        Glide.with(getContext()).load(profilePhoto.getUrl())
+                .fitCenter() // scale image to fill the entire ImageView
+                .transform(new RoundedCorners(radius))
+                .into(ivProfileView);
+    }
+
+    private void setupSaveChanges(HashMap<String, MyPair<EditText, Boolean>> editUserFields) {
+        MyPair<EditText, Boolean> pair;
+        for (String key : editUserFields.keySet()) {
+            pair = editUserFields.get(key);
+            if (pair.second) {
```

Comment:
do we actually need "pair.second"? could we just check if the edit text is empty or not?